### ISSUE TITLE
problem with place with map info in gedcom

### ISSUE THREAD

**Issue by Joppla** [dkniffin/webtrees-openstreetmap#22]
If there is map-info in the gedcom file (nr 3 and 4):
```
1 DEAT
2 DATE 18 APR 1991
2 PLAC Deventer, Overijssel, Nederland
3 MAP
4 LATI 52.251663
4 LONG 6.154833
```
 the place is shown on 0.0000 - 0.0000. 
![afbeelding](https://cloud.githubusercontent.com/assets/25467829/22486162/b84964fc-e808-11e6-8543-9a38431018f8.png)

If I remove 3 and 4 than the place is shown correct, probably according to the google maps place hierarchy.

**Comment by dkniffin** [comment]
Hmm, interesting. I haven't done much development on this library recently, so I probably won't get time to fix this anytime soon, but if you submit a PR, I'll review it and merge it.

**Comment by Joppla** [comment]
Hi, thanks for your reaction, but what is a PR?

**Comment by dkniffin** [comment]
Oh, sorry. It's [a pull request](https://help.github.com/articles/about-pull-requests/). Essentially, if you can make the change to the code, and send them to me (usually done via a PR, but I'd also be fine with you just leaving them in a comment here), I'll merge them into the master branch, to be included in the next release.

**Comment by Joppla** [comment]
Ok, I'm afraid that I'm more a user than a programmer. I will look if I understand anything from the code, but I m afraid that I won't be able to find out how the code should be. The same for the other issue I reported yesterday.

**Comment by dkniffin** [comment]
Ok, no worries. I'll probably get to this eventually, just wanted to let you take a stab at it if you wanted to get it done sooner. If you do decide to do one, I'd guess that #21 would probably be easier.

**Comment by dkniffin** [comment]
Also, I think this issue is somewhere in here: https://github.com/dkniffin/webtrees-openstreetmap/blob/master/classes/FactPlace.php#L81

**Comment by Joppla** [comment]
I will look, what I can understand from the code this evening :-)

**Comment by dkniffin** [comment]
Ok, sounds good. It's written in [PHP](http://www.w3schools.com/php/php_intro.asp), which isn't the best language (imo), but it's what webtrees is built on, so it's what we're using :P

**Comment by Joppla** [comment]
If I // line 88 return, then it works correct. It must be something with the interpretation of the data from the GedCom file. I will look further later.

**Comment by Joppla** [comment]
line [#82](https://github.com/dkniffin/webtrees-openstreetmap/blob/master/classes/FactPlace.php#L82) and #83 returns "int (1)" if there is a LATI or LONG and "int (0)" if there isn't. So it seems
`			$gedcom_lat = preg_match("/\d LATI (.*)/", $fact->getGedcom(), $match1);`
The needed value is in var $match1:
`array(2) {
  [0]=>
  string(17) "4 LATI N52.251663"
  [1]=>
  string(10) "N52.251663"`
Maybe you have now enough to rewrite the code?

**Comment by dkniffin** [comment]
Oh, yep, you're right. Try  changing that block of code to:

```
// First look to see if the lat/lon is hardcoded in the gedcom
preg_match("/\d LATI (.*)/", $fact->getGedcom(), $gedcom_lat);
preg_match("/\d LONG (.*)/", $fact->getGedcom(), $gedcom_lon);
if ($gedcom_lat && $gedcom_lon) {
	// If it's hardcoded, we're done.
	$this->lat = $gedcom_lat;
	$this->lon = $gedcom_lon;
	return;
}
```

**Comment by Joppla** [comment]
I have to go home, I will test this evening further :-) Thanks for now for the coorporation!

**Comment by dkniffin** [comment]
👍  Thanks for looking into it!

**Comment by Joppla** [comment]
Hello,
The $gedcom_lat and $gedcom_lon are arrays. Your code gives an empty screen. But this code works in my webtrees:
`		if (!$fact->getPlace()->isEmpty()) {
			// First look to see if the lat/lon is hardcoded in the gedcom
			$gedcom_lat = preg_match("/\d LATI (.*)/", $fact->getGedcom(), $match_lati);
			$gedcom_lon = preg_match("/\d LONG (.*)/", $fact->getGedcom(), $match_long);
			if ($gedcom_lat && $gedcom_lon) {
				// If it's hardcoded, we're done.
				$this->lat = $match_lati[1];
				$this->lon = $match_long[1];
				return;
			}`

**Comment by dkniffin** [comment]
Ah, right. Ok, cool. I'll add this to the codebase later today or tomorrow. Thanks for looking into this for me!

**Comment by dkniffin** [comment]
Oh, I just realized this issue was fixed by #18, but I haven't bumped the version yet. 🤦‍♂️ I'll do that tomorrow.
